### PR TITLE
common practices

### PR DESCRIPTION
**Pull request by thibaudk**
First draft presenting common practices.
More to come, 
Sorry for the probable spelling errors.
Is this "tutorial" like format the right one for this section of the doc ?

### CODE REVIEW

File: docs/reference-manual/common-practices/looping.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+tag: "Presentation"
+visible: true
+---
+
+# Looping
+
+## Process Loops
+
+Time based processes like automations, media and sub-scenarios can be looped for as long as their parent interval is active. When selecting a process, eiher form the object window or by clicking on the it's upper margin, the loop option will appear in the `Inspector`.
```
Comment (jcelerier) [inline docs/reference-manual/common-practices/looping.md:24]: Time-based processes such as automations, media and sub-scenarios can be looped for as long as their parent interval is active. When selecting a process, either from the object panel **(TODO mettre lien vers page correspondante de la doc)** or by clicking on its upper margin **(est-ce qu'on aurait pas un meilleur mot?)**, the loop option will appear in the `Inspector`

File: docs/reference-manual/common-practices/looping.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+![Process looping](/score-docs/assets/images/reference-manual/common-practices/loopProcess.gif "Process looping")
+
+## Loop with transitions
+
+To enable the repetition of certain parts of a scenario, we use transitions to go "back in time". Transitions are instantaneous intervals that can connect two states ragrdless of their chonological relasionship.
```
Comment (jcelerier) [inline docs/reference-manual/common-practices/looping.md:30]: regardless of their chronological relationship.

File: docs/reference-manual/common-practices/looping.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+![Process looping](/score-docs/assets/images/reference-manual/common-practices/loopProcess.gif "Process looping")
+
+## Loop with transitions
+
+To enable the repetition of certain parts of a scenario, we use transitions to go "back in time". Transitions are instantaneous intervals that can connect two states ragrdless of their chonological relasionship. 
+
+![Loop with Transitions](/score-docs/assets/images/reference-manual/common-practices/loopStructure.gif "Loop with transitions")
+
+It is important to note however that in this case, the loop will go on forever as well as playing and repalying all folowing intervals.
```
Comment (jcelerier) [inline docs/reference-manual/common-practices/looping.md:34]: Juste "In this case, the loop will go on forever as well as playing and replaying all following intervals." suffit

File: docs/reference-manual/common-practices/looping.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+## Nested loops
+
+For more control and to avoid prpetual looping and branshing, it may be necessary to isolate a repeating structure in an sub-scenario. Triggering the end of the interval containing the sub-scenario will ensure the stopping of the loop as well as the execution of the folowing intervals.
```
Comment (jcelerier) [inline docs/reference-manual/common-practices/looping.md:38]: - prpetual => perpetual
- branshing => branching
-  in a sub-scenario
- will ensure the stopping of the loop => will ensure that the loop stops
- folowing => following

File: docs/reference-manual/common-practices/looping.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+![Nested Loops](/score-docs/assets/images/reference-manual/common-practices/nestedLoop.gif "Nested loop")
+
+## Repetition ammount
+
+Adding a trigger on the last `Sync` of a loop and setting it's maximum duration allows to control the amount of repetition. The trigger left by default to be "nerver" true will still be vaildated when the maximum duration of the precceding interval has been reached. With this metthod, the time spent looping is equal to the loop duration + the maximum duration, and can be fine tuned by manipulating the closing parenthesis.
```
Comment (jcelerier) [inline docs/reference-manual/common-practices/looping.md:44]: - setting it's maximum duration => setting its maximum duration
- nerver => never
- precceding => preceding
- metthod => method
-  fine tuned  => fine-tuned

File: docs/reference-manual/common-practices/looping.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+![epetition ammount](/score-docs/assets/images/reference-manual/common-practices/repetition-ammount.gif "repetition ammount")
```
Comment (jcelerier) [inline docs/reference-manual/common-practices/looping.md:46]: epetition ammount => repetition amount 
(les deux fois)

File: docs/reference-manual/common-practices/looping.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+## Loop branches 
+
+Transitions actually connect to `Syncs` (vertical doted lines), therefore, transitioning to a specific `Sync` will execute or re-execute all connected branches. When sevral treansitions are connected to the same `Sync`, the smallest loop will then always restart first and cut short all parrallel intervals.
```
Comment (jcelerier) [inline docs/reference-manual/common-practices/looping.md:50]: - sevral => several
- treansitions => transitions
- smallest => shortest ? 
- parrallel => parallel

File: docs/reference-manual/common-practices/looping.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+![Loop branches](/score-docs/assets/images/reference-manual/common-practices/loopBranches.gif "Loop branches")
+
+Transitions can provied a workaround in this situation. Since transitions are instantaneous intervals, they can also be used to isolate parllel loops without interfering with timming and their original synchronisation.
```
Comment (jcelerier) [inline docs/reference-manual/common-practices/looping.md:54]: - provied => provide
- parllel => parallel
- timming => timing

File: docs/reference-manual/common-practices/common-practices.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+has_children: true
+
+parent: Reference manual
+
+permalink: /docs/reference-manual/common-practices
+---
+
+# Common practices
+
+The Following pages will guide you through reocuring patterns and snipets that can be useful in many different use-cases.
```
Comment (jcelerier) [inline docs/reference-manual/common-practices/common-practices.md:14]: - reocuring => reoccuring
- snipets => snippets
-

File: docs/reference-manual/common-practices/common-practices.md [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+Enjoy advanced scenario authoring for your intermedia creation !
```
Comment (jcelerier) [inline docs/reference-manual/common-practices/common-practices.md:16]: hehe

File: docs/reference-manual/common-practices/looping.md (hunk 1 above)
Comment (thibaudk) [reply docs/reference-manual/common-practices/looping.md:24]: upper margin (est-ce qu'on aurait pas un meilleur mot?)
"input bar" ? "process header" ? "process thumbnail"

Comment (thibaudk) [reply docs/reference-manual/common-practices/looping.md:24]: (TODO mettre lien vers page correspondante de la doc)
Maybe this section could be fleshed out and isolated as it's own section in the "Quick start" or in the "Reference manual" "Process" page ?

Comment (jcelerier) [reply docs/reference-manual/common-practices/looping.md:24]: - process header
- Maybe this section could be fleshed out and isolated as it's own section in the "Quick start" or in the "Reference manual" "Process" page ? => yes :) but in a further work, let's merge as-is

File: docs/reference-manual/common-practices/looping.md (hunk 7 above)
Comment (thibaudk) [reply docs/reference-manual/common-practices/looping.md:50]: smallest => shortest ?
ia am using "cut short" just after

File: docs/reference-manual/common-practices/common-practices.md (hunk 9 above)
Comment (thibaudk) [reply docs/reference-manual/common-practices/common-practices.md]: done in f0feab7

File: docs/reference-manual/common-practices/looping.md (hunk 8 above)
Comment (thibaudk) [reply docs/reference-manual/common-practices/looping.md]: done in f0feab7

File: docs/reference-manual/common-practices/looping.md (hunk 6 above)
Comment (thibaudk) [reply docs/reference-manual/common-practices/looping.md]: done in f0feab7

File: docs/reference-manual/common-practices/looping.md (hunk 5 above)
Comment (thibaudk) [reply docs/reference-manual/common-practices/looping.md]: done in f0feab7

File: docs/reference-manual/common-practices/looping.md (hunk 4 above)
Comment (thibaudk) [reply docs/reference-manual/common-practices/looping.md]: done in f0feab7

File: docs/reference-manual/common-practices/looping.md (hunk 3 above)
Comment (thibaudk) [reply docs/reference-manual/common-practices/looping.md]: done in f0feab7

File: docs/reference-manual/common-practices/looping.md (hunk 2 above)
Comment (thibaudk) [reply docs/reference-manual/common-practices/looping.md]: done in f0feab7

**Comment by jcelerier** [comment]
looks good ! thanks !

**Pull request merged**